brave/brave-core
# Refactor Brave Ads global state.

Move Brave Ads manager classes global state to AdsGlobalState.

Resolves https://github.com/brave/brave-browser/issues/29657



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

- Check that an ad cannot be served with ad features enabled from the command line (i.e. `--enable-features=EpsilonGreedyBandit`)
- Check that an ad can be served with ad features enabled from the command line with staging rewards (i.e. `--enable-features=EpsilonGreedyBandit --rewards=staging=true` )
- Check that confirmations contains buildChannel and rotating_hash
